node/reqres.js

```diff
@@ -122,18 +130,25 @@ function TChannelOutgoingResponse(id, options, sendFrame) {
     self.arg2 = options.arg2 || emptyBuffer;
     self.arg3 = options.arg3 || emptyBuffer;
     self.sendFrame = sendFrame;
+    self.sent = false;
 }
 
 inherits(TChannelOutgoingResponse, EventEmitter);
 
 TChannelOutgoingResponse.prototype.send = function send(err, res1, res2) {
     var self = this;
-    if (err) {
-        self.ok = false;
-        var errArg = isError(err) ? err.message : JSON.stringify(err); // TODO: better
-        self.sendFrame(self.name, res1, errArg);
+    if (self.sent) {
+        self.emit('error', new Error('response already sent'));
     } else {
-        self.sendFrame(self.name, res1, res2);
+        self.sent = true;
+        if (err) {
+            self.ok = false;
+            var errArg = isError(err) ? err.message : JSON.stringify(err); // TODO: better
```

Comment:
no this is application error only, the error frame is protocol level error exclusively.